flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -153,6 +176,143 @@ public static ZooKeeperLeaderElectionService createLeaderElectionService(
 	}
 
 	/**
+	 * Creates a {@link ZooKeeperSubmittedJobGraphs} instance.
+	 *
+	 * @param client        The {@link CuratorFramework} ZooKeeper client to use
+	 * @param configuration {@link Configuration} object
+	 * @return {@link ZooKeeperSubmittedJobGraphs} instance
+	 */
+	public static ZooKeeperSubmittedJobGraphs createSubmittedJobGraphs(
+			CuratorFramework client,
+			Configuration configuration) throws Exception {
+
+		checkNotNull(configuration, "Configuration");
+
+		// State backend
+		String stateBackend = configuration.getString(
+				ConfigConstants.STATE_BACKEND,
+				ConfigConstants.DEFAULT_STATE_BACKEND);
+
+		if (!stateBackend.toLowerCase().equals("filesystem")) {
```

Comment:
Could we use the enum `StateBackend` here rather than string equality? Maybe it would be a good idea to use a factory method for the state backend depending on the `StateBackend` enum value.